lib/poller.js

```diff
@@ -150,37 +189,66 @@ class Poller {
   }
 
   /**
-   * Checks if secret exists, if not trigger a poll
+   * Checks status of external secret and determines time to next poll
+   * If current observed generation is older than ES generation it will poll right away
+   * otherwise check when it was last polled and set timeout for next poll
    */
-  async _checkForSecret () {
-    const kubeNamespace = this._kubeClient.api.v1.namespaces(this._namespace)
-
+  async _checkExternalSecretStatus () {
```

Comment:
naming nit: how about `_scheduleNextPoll`?  I think that more accurately describes what this function does.